### PR TITLE
12755 - Unhandled SystemExit: 1 and various performance fixes

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*
https://github.com/bcgov/entity/issues/12755

*Description of changes:*
Performance fixes + Circular reference issue fix.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by ritvick — APPROVED** [review]

**Review by ritvick — APPROVED** [review]

**Comment by seeker25** [comment]
There's 3 more cases to go for this, I'll get you to review again I think.

File: auth-api/src/auth_api/resources/org.py [hunk 1]
```diff
@@ -504,7 +504,8 @@ def get(org_id):
             invitations = InvitationService.get_invitations_for_org(org_id=org_id,
                                                                     status=invitation_status)
 
-            response, status = {'invitations': InvitationSchema().dump(invitations, many=True)}, http_status.HTTP_200_OK
+            response, status = {'invitations': InvitationSchema(exclude=['membership.org'])
+                                .dump(invitations, many=True)}, http_status.HTTP_200_OK
```
[the remark this answers is not in the document]
Comment (seeker25) [reply auth-api/src/auth_api/resources/org.py:508]: The schemas exclude all of the PII data. It's mostly email, createdBy.. that's about it.

File: auth-api/src/auth_api/schemas/base_schema.py [hunk 2]
```diff
@@ -33,22 +33,19 @@ class Meta:  # pylint: disable=too-few-public-methods
     modified_by = fields.Function(
         lambda obj: f'{obj.modified_by.firstname} {obj.modified_by.lastname}' if obj.modified_by else None
     )
+    exclude = ('versions')
 
     @post_dump(pass_many=True)
     def _remove_empty(self, data, many):
         """Remove all empty values and versions from the dumped dict."""
         if not many:
-            for key in list(data):
```
Comment (seeker25) [inline auth-api/src/auth_api/schemas/base_schema.py:41]: This code would exclude the field, after it's already been queried.

File: auth-api/src/auth_api/schemas/base_schema.py [hunk 3]
```diff
@@ -33,22 +33,19 @@ class Meta:  # pylint: disable=too-few-public-methods
     modified_by = fields.Function(
         lambda obj: f'{obj.modified_by.firstname} {obj.modified_by.lastname}' if obj.modified_by else None
     )
+    exclude = ('versions')
```
Comment (seeker25) [inline auth-api/src/auth_api/schemas/base_schema.py:36]: Excluding it in the base schema here, that way versions isn't queried when an object is serialized.

File: auth-api/src/auth_api/resources/org.py [hunk 4]
```diff
@@ -504,7 +505,8 @@ def get(org_id):
             invitations = InvitationService.get_invitations_for_org(org_id=org_id,
                                                                     status=invitation_status)
 
-            response, status = {'invitations': InvitationSchema().dump(invitations, many=True)}, http_status.HTTP_200_OK
+            response, status = {'invitations': InvitationSchema(exclude=['membership.org'])
```
Comment (seeker25) [inline auth-api/src/auth_api/resources/org.py:508]: Follows the same pattern as Line 411.

File: auth-api/migrations/alembic.ini [hunk 5]
```diff
@@ -34,6 +34,11 @@ level = INFO
 handlers =
 qualname = alembic
 
+[logger_flask_migrate]
```
Comment (seeker25) [inline auth-api/migrations/alembic.ini:37]: Without this, it's hard to see any migration errors.

File: auth-api/src/auth_api/models/contact_link.py [hunk 6]
```diff
@@ -30,10 +30,10 @@ class ContactLink(VersionedModel):  # pylint: disable=too-few-public-methods
     __tablename__ = 'contact_links'
 
     id = Column(Integer, primary_key=True)
-    contact_id = Column(Integer, ForeignKey('contacts.id'))
-    entity_id = Column(Integer, ForeignKey('entities.id'))
-    user_id = Column(Integer, ForeignKey('users.id'))
-    org_id = Column(Integer, ForeignKey('orgs.id'))
+    contact_id = Column(Integer, ForeignKey('contacts.id'), index=True)
```
Comment (seeker25) [inline auth-api/src/auth_api/models/contact_link.py:33]: I found these suggestions, by running a database profiling tool.

File: auth-api/src/auth_api/schemas/base_schema.py [hunk 7]
```diff
@@ -13,14 +13,21 @@
...
 from auth_api.models import ma
+from auth_api.models.base_model import VersionedModel
 
 
 class BaseSchema(ma.ModelSchema):  # pylint: disable=too-many-ancestors
     """Base Schema."""
 
+    def __init__(self, *args, **kwargs):
+        """Excludes versions."""
+        if issubclass(self.opts.model, VersionedModel):
```
Comment (seeker25) [inline auth-api/src/auth_api/schemas/base_schema.py:27]: Exclude versions, it's another HUGE table that is only used for audit.

File: auth-api/migrations/alembic.ini (hunk 5 above)
Comment (ritvick) [reply auth-api/migrations/alembic.ini:37]: +1

**Review by ritvick — APPROVED** [review]

**Pull request merged**